request-manager/graph/resolver.go

```diff
@@ -121,193 +123,168 @@ func (o *resolver) RequestArgs(jobArgs map[string]interface{}) ([]proto.RequestA
+				conditional, err := chooseConditional(nodeSpec, jobArgsCopy)
 				if err != nil {
-					return nil, fmt.Errorf("in seq %s, node %s: %s", sequenceName, n.Name, err)
+					return nil, fmt.Errorf("in seq %s, node %s: %s", sequenceName, nodeSpec.Name, err)
 				}
 				cfg := buildSequenceConfig{
-					wrapperName:       "conditional_" + n.Name,
+					graphName:         "conditional_" + nodeSpec.Name,
 					sequenceName:      conditional,
 					jobArgs:           jobArgsCopy,
-					sequenceRetry:     n.Retry,
-					sequenceRetryWait: n.RetryWait,
+					sequenceRetry:     nodeSpec.Retry,
+					sequenceRetryWait: nodeSpec.RetryWait,
 				}
-				reqSeqGraph, err = o.buildSequence(cfg)
+				reqSubgraph, err = r.buildSequence(cfg)
 				if err != nil {
-					return nil, fmt.Errorf("in seq %s, node %s: %s", sequenceName, n.Name, err)
+					return nil, fmt.Errorf("in seq %s, node %s: %s", sequenceName, nodeSpec.Name, err)
 				}
-			} else if n.IsSequence() {
-				// Node is a sequence, recursively construct its components
+			} else if nodeSpec.IsSequence() {
+				// Node is a sequence: recursively build the subgraph
 				cfg := buildSequenceConfig{
-					wrapperName:       "sequence_" + n.Name,
-					sequenceName:      *n.NodeType,
+					graphName:         "sequence_" + nodeSpec.Name,
+					sequenceName:      *nodeSpec.NodeType,
 					jobArgs:           jobArgsCopy,
-					sequenceRetry:     n.Retry,
-					sequenceRetryWait: n.RetryWait,
+					sequenceRetry:     nodeSpec.Retry,
+					sequenceRetryWait: nodeSpec.RetryWait,
 				}
-				reqSeqGraph, err = o.buildSequence(cfg)
+				reqSubgraph, err = r.buildSequence(cfg)
 				if err != nil {
-					return nil, fmt.Errorf("in seq %s, node %s: %s", sequenceName, n.Name, err)
+					return nil, fmt.Errorf("in seq %s, node %s: %s", sequenceName, nodeSpec.Name, err)
 				}
 			} else {
-				// Node is a job, create a graph that contains only the node
-				reqSeqGraph, err = o.buildSingleVertexGraph(n, jobArgsCopy)
+				// Node is a job: create the proto.Job and put
+				// it in a graph
+				reqSubgraph, err = r.buildSingleVertexGraph(nodeSpec, jobArgsCopy)
 				if err != nil {
-					return nil, fmt.Errorf("in seq %s, node %s: cannot build job: %s", sequenceName, n.Name, err)
+					return nil, fmt.Errorf("in seq %s, node %s: cannot build job: %s", sequenceName, nodeSpec.Name, err)
 				}
 			}
-			if !reqSeqGraph.IsValidGraph() {
-				return nil, fmt.Errorf("malformed request sequence graph created for %s", seqGraphNode.Name)
-			}
 
-			// Add the new node to the map of completed components
-			components = append(components, reqSeqGraph)
+			components = append(components, reqSubgraph)
 
-			// If the node (or sequence) was determined to set any args
+			// If the node or sequence was determined to set any args
 			// copy them from jobArgsCopy into the main jobArgs
-			err = setNodeArgs(n, jobArgs, jobArgsCopy)
+			err = setNodeArgs(nodeSpec, jobArgs, jobArgsCopy)
 			if err != nil {
```

Comment:
`if err := setNodeArgs(nodeSpec, jobArgs, jobArgsCopy); err != nil {`

Whenever the line is short enough, the `if err := func(); err != nil {` is a hallmark of idiomatic Go. Also, reusing `err` here is technically fine but imho it's a good practice to avoid for same reason we avoid global vars: the more locally scoped the better to avoid any chances of "cross-contamination".